BinaryTreeTest.java

```diff
@@ -9,63 +15,71 @@
  * Created by adi on 6/1/2017.
  */
 public class BinaryTreeTest {
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private BinaryTree<Integer> myTree;
+    @Before
+    public void setUp(){
+        System.setOut(new PrintStream(outContent));

```

Comment:
I would advise against testing output of the class unless there are strict requirements about output format. Why not simplify things and just test whether collection returned from findAncestors contains the right elements?

Errors can safely be tested by just checking for null (which you already do in some tests). If you want to be extra safe, you could use exceptions with a separate class (or one class with different error codes/messages) to signal them and check if correct exception is thrown.